src/index/IndexBuilderMain.cpp

```diff
@@ -152,8 +149,8 @@ int main(int argc, char** argv) {
     return EXIT_FAILURE;
   }
   if (stxxlMemoryGB.has_value()) {
-    index.stxxlMemoryInBytes() =
-        1024ul * 1024ul * 1024ul * stxxlMemoryGB.value();
+    index.stxxlMemory() = ad_utility::MemorySize::bytes(
+        1024UL * 1024UL * 1024UL * stxxlMemoryGB.value());
```

Comment:
Also directly make the corresponding `program_options()` a `MemorySize`, like in the `ServerMain`